### PR TITLE
syslog: add syslog option definition

### PR DESCRIPTION
**Pull request by zhhyu7**
## Summary
We will integrate network management tools in future projects. In the process of porting the tools, we encounter some situations where the structure is not defined, or the returned data types do not match the expectations. Refer to the common implementation of other systems and add relevant definitions.
## Testing
sim:local

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Comment by jerpelea** [comment]
please add commit message describing the change

File: include/syslog.h [hunk 1]
```diff
@@ -162,9 +169,11 @@ extern "C"
  *
  ****************************************************************************/
 
-#if 0 /* Not supported */
-void openlog(FAR const char *ident, int option, int facility);
-#endif
+/* Not supported */
+
+static inline void openlog(const char *ident, int option, int facility)
```
Comment (xiaoxiang781216) [inline include/syslog.h:174]: let's change to empty function instead inline to support more compiler

File: include/syslog.h [hunk 2]
```diff
@@ -176,9 +183,9 @@ void openlog(FAR const char *ident, int option, int facility);
  *
  ****************************************************************************/
 
-#if 0 /* Not supported */
+/* Not supported */
```
Comment (xiaoxiang781216) [inline include/syslog.h:186]: ditto

File: libs/libc/syslog/lib_syslog.c [hunk 3]
```diff
@@ -35,6 +35,18 @@
  * Public Functions
  ****************************************************************************/
 
+/* Not supported */
```
Comment (xiaoxiang781216) [inline libs/libc/syslog/lib_syslog.c:38]: remove

File: libs/libc/syslog/lib_syslog.c [hunk 4]
```diff
@@ -35,6 +35,18 @@
...
+
+void openlog(FAR const char *ident, int option, int facility)
+{
+}
+
+/* Not supported */
```
Comment (xiaoxiang781216) [inline libs/libc/syslog/lib_syslog.c:44]: remove

**Comment by zhhyu7** [comment]
@xiaoxiang781216 , I went back to the implementation of the macro definition, it looks like there are no compilation issues now. please review, thanks.

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/syslog.h [hunk 5]
```diff
@@ -162,9 +169,8 @@ extern "C"
  *
  ****************************************************************************/
 
-#if 0 /* Not supported */
-void openlog(FAR const char *ident, int option, int facility);
-#endif
+/* Not supported */
+#define openlog(i, o, f) {(void)(i); (void)(o); (void)(f);}
```
Comment (pkarashchenko) [inline include/syslog.h:173]: `UNUSED` instead of `(void)`?

**Pull request merged**